sycl/include/sycl/backend.hpp

```diff
@@ -213,7 +213,78 @@ get_native<backend::ext_oneapi_cuda, device>(const device &Obj) {
   return static_cast<backend_return_t<backend::ext_oneapi_cuda, device>>(
       Obj.getNative());
 }
-#endif
+
+#ifndef SYCL_EXT_ONEAPI_BACKEND_CUDA_EXPERIMENTAL
+template <>
+__SYCL_DEPRECATED(
+    "Context interop is deprecated for CUDA. If a native context is required,"
+    " use cuDevicePrimaryCtxRetain with a native device")
+inline backend_return_t<backend::ext_oneapi_cuda, context> get_native<
+    backend::ext_oneapi_cuda, context>(const context &Obj) {
+  if (Obj.get_backend() != backend::ext_oneapi_cuda) {
+    throw sycl::exception(make_error_code(errc::backend_mismatch),
+                          "Backends mismatch");
+  }
+  return reinterpret_cast<backend_return_t<backend::ext_oneapi_cuda, context>>(
+      Obj.getNative());
+}
+
+template <>
+__SYCL_DEPRECATED("Context interop is deprecated for CUDA. If a native context"
+                  " is required, use cuDevicePrimaryCtxRetain with a native"
+                  " device")
+inline backend_return_t<backend::ext_oneapi_cuda, context> interop_handle::
+    get_native_context<backend::ext_oneapi_cuda>() const {
```

Comment:
Should we keep this interface? This one doesn't take a context and is only available in a host task, where we know the device so we can return the proper CUDA/HIP context.

Reply:
@hdelan, ping.

Reply:
Thanks changes made